docroot/sites/all/modules/contrib/user_cancel_immediate/user_cancel_immediate.module

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/**
+ * @file
+ * Module file for user_cancel_immediate
+ * Change user cancellation to be immmediate, without any email confirmation.
+ *
+ * Thanks to Clive, http://drupal.stackexchange.com/a/32427/449
+ */
+
+
+/**
+ * Implements hook_form_FORMID_alter()
+ * Adds a custom submit handler
+ */
+function user_cancel_immediate_form_user_cancel_confirm_form_alter(&$form, &$form_state, $form_id) {
+  $form['#submit'][] = '_user_cancel_immediate_user_cancel_form_submit';
```

Comment:
Ignore that last part, I see user_cancel_confirm_form_submit assumes (incorrectly) that if the value is false the user must be an admin.